### DISCUSSION TITLE
Need help in resolving  'Invalid Compact JWE' error in Next Auth

### DISCUSSION THREAD

**Discussion by ChiragAgg5k** [nextauthjs/next-auth#8626] [Help] [answered]
I am trying to make next auth work with Neon's postgres database using Prisma and include Google's OAuth. So far i have done everything, and all the data of the user is successfully being stored in the database. But when trying to fetch the data using `getServerSession` method, i get this error:

```
[next-auth][error][JWT_SESSION_ERROR] 
https://next-auth.js.org/errors#jwt_session_error Invalid Compact JWE {
  message: 'Invalid Compact JWE',
  stack: 'JWEInvalid: Invalid Compact JWE\n' +
    '    at compactDecrypt (webpack-internal:///(rsc)/./node_modules/jose/dist/node/cjs/jwe/compact/decrypt.js:18:15)\n' +
    '    at jwtDecrypt (webpack-internal:///(rsc)/./node_modules/jose/dist/node/cjs/jwt/decrypt.js:10:61)\n' +
    '    at Object.decode (webpack-internal:///(rsc)/./node_modules/next-auth/jwt/index.js:44:52)\n' +
    '    at async Object.session (webpack-internal:///(rsc)/./node_modules/next-auth/core/routes/session.js:25:34)\n' +
    '    at async AuthHandler (webpack-internal:///(rsc)/./node_modules/next-auth/core/index.js:161:37)\n' +
    '    at async getServerSession (webpack-internal:///(rsc)/./node_modules/next-auth/next/index.js:125:21)\n' +
    '    at async Page (webpack-internal:///(rsc)/./app/(pages)/dashboard/page.tsx:14:21)',
  name: 'JWEInvalid'
}
```

So far i only found one thread regarding this [here](https://github.com/nextauthjs/next-auth/discussions/4255), where although the error is same the stack is different. And the solution proposed in it, which was to clear cookies, did not work.

Here's my next-auth api file:

```ts
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import {env} from "@/env";
import { PrismaAdapter } from "@auth/prisma-adapter";
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

const authOptions = {
    adapter: PrismaAdapter(prisma),
    providers: [
        GoogleProvider({
            clientId: env.GOOGLE_CLIENT_ID,
            clientSecret: env.GOOGLE_CLIENT_SECRET
        }),
    ],
    secret: env.NEXTAUTH_SECRET,
    strategy: "jwt",
}

const handler = NextAuth(authOptions);
export { handler as GET, handler as POST };
```

Also i am only getting this error with `getServerSession`. The `useSession` hook works fine.

**Comment by luannguyenQV** [comment, marked as answer]
@ChiragAgg5k You can try this one:
`  const session = await getServerSession(authConfigs);`

**Reply by kevinhellos** [reply]
Thank you for this. This is working.

**Comment by achmadfha** [comment]
i have solved this problem

try like this
`const session = await getServerSession(authOptions) `
remember using async on your functions

**Comment by azeddine-hmd** [comment]
i had the same problem and it worked when i supplied authOptions. but, error [next-auth][error][JWT_SESSION_ERROR] still presists.
